GUI/Types/Renderer/HitboxSetSceneNode.cs

```diff
@@ -0,0 +1,202 @@
+            };
+        }
+
+        public void SetHitboxSet(string set)
+        {
+            if (set == null)
+            {
+                currentSet = null;
+                return;
+            }
+
+            currentSet = hitboxSets[set];
+        }
+
+        public override void Update(Scene.UpdateContext context)
+        {
+            if (currentSet == null)
+            {
+                return;
+            }
+
+            LocalBoundingBox = new AABB(new Vector3(float.MinValue), new Vector3(float.MaxValue));
+
+            Frame frame = null;
+
+            if (animationController.ActiveAnimation != null)
+            {
+                if (animationController.IsPaused)
+                {
+                    frame = animationController.FrameCache.GetFrame(animationController.ActiveAnimation, animationController.Frame);
+                }
+                else
+                {
+                    frame = animationController.FrameCache.GetInterpolatedFrame(animationController.ActiveAnimation, animationController.Time);
+                }
+            }
+
+            foreach (var root in skeleton.Roots)
+            {
+                GetAnimationMatrixRecursive(root, Matrix4x4.Identity, frame);
+            }
+        }
+
+        private void GetAnimationMatrixRecursive(Bone bone, Matrix4x4 bindPose, Frame frame)
+        {
+            if (frame != null)
+            {
+                var transform = frame.Bones[bone.Index];
+                bindPose = Matrix4x4.CreateScale(transform.Scale)
+                    * Matrix4x4.CreateFromQuaternion(transform.Angle)
+                    * Matrix4x4.CreateTranslation(transform.Position)
+                    * bindPose;
+            }
+            else
+            {
+                bindPose = bone.BindPose * bindPose;
+            }
+
+            boneMatrices[bone.Index] = bindPose;
+
+            foreach (var child in bone.Children)
+            {
+                GetAnimationMatrixRecursive(child, bindPose, frame);
+            }
+        }
+
+        private void RenderHitboxSet(Scene.RenderContext context, HitboxSetData hitboxSetData)
+        {
+            var hitboxSet = hitboxSetData.HitboxSet;
+            for (var i = 0; i < hitboxSet.Length; i++)
+            {
+                var shape = hitboxSetData.SceneNodes[i];
+                var hitbox = hitboxSet[i];
+                var boneId = hitboxSetData.HitboxBoneIndexes[i];
+                var targetTransform = boneId == -1 ? Matrix4x4.Identity : boneMatrices[boneId];
+
+                if (hitbox.TranslationOnly)
+                {
+                    Matrix4x4.Decompose(targetTransform, out _, out _, out var translation);
+                    shape.Transform = Matrix4x4.CreateTranslation(translation);
+                }
+                else
+                {
+                    shape.Transform = targetTransform;
```

Comment:
Transform updates should be handled in Update, so this loop only manages rendering.